src/diskreader.cxx

```diff
@@ -93,6 +93,23 @@ int DiskReader::loadPreferences()
 			LUPPP_NOTE("No default controllers active.");
 		}
 
+		cJSON* hotkeyAssignment = cJSON_GetObjectItem(preferencesJson, "hotkeyAssignment");
+        if (hotkeyAssignment) {
+            keyToGrid.clear();
+            cJSON* key = hotkeyAssignment->child;
+            while (key) {
+                int keycode = std::stoi(key->string);
```

Comment:
is there any way this can throw an exception due to invalid/non-int-convertable input? `stoi` could[1], so this needs a try {} catch to avoid taking the whole program down on int-convert-error?

Perhaps the `C` lang `atoi` (with no exceptions, and no error handling.. returning `0` instead) is better as it won't throw exceptions?

[1] https://en.cppreference.com/w/cpp/string/basic_string/stol